packages/get-platform/src/getPlatform.ts

```diff
@@ -102,46 +106,47 @@ export function parseOpenSSLVersion(input: string): string | undefined {
   return
 }
 
-// getOpenSSLVersion returns the OpenSSL version excluding the patch version, e.g. "1.1.x"
-export async function getOpenSSLVersion(): Promise<string | undefined> {
-  const [version, ls] = await Promise.all([
-    gracefulExec(`openssl version -v`),
-    gracefulExec(`
-      ls -l /lib64 | grep ssl;
-      ls -l /usr/lib64 | grep ssl;
-    `),
-  ])
-
-  if (version) {
-    const v = parseOpenSSLVersion(version)
-    if (v) {
-      return v
-    }
-  }
+type GetOpenSSLVersionParams = {
+  arch: Arch
+  distro: GetOSResult['distro']
+}
 
-  if (ls) {
-    const match = /libssl\.so\.(\d+\.\d+)\.\d+/.exec(ls)
+/**
+ * On Linux, returns the OpenSSL version excluding the patch version, e.g. "1.1.x".
+ * Reading the version from the libssl.so file is more reliable than reading it from the openssl binary.
+ * This function never throws.
+ */
+export async function getSSLVersion(args: GetOpenSSLVersionParams): Promise<string | undefined> {
+  const libsslVersion: string | undefined = await match(args)
+    .with({ distro: 'musl' }, () => {
+      /* Linux Alpine */
+      return getFirstSuccessfulExec(['ls -l /lib/libssl.so.3', 'ls -l /lib/libssl.so.1.1'])
+    })
+    .otherwise(() => {
+      return getFirstSuccessfulExec(['ls -l /lib64 | grep ssl', 'ls -l /usr/lib64 | grep ssl'])
+    })
+
+  if (libsslVersion) {
+    const match = /libssl\.so\.(\d+\.\d+)\.\d+/.exec(libsslVersion)
```

Comment:
I think this regex won't match the filename on Alpine

Reply:
I was just looking into that, running things locally